gson/src/org/immutables/gson/stream/JsonParserReader.java

```diff
@@ -141,7 +144,7 @@ public String nextName() throws IOException {
   public String nextString() throws IOException {
     requirePeek();
     if (!isLenient()) {
-      expect(VALUE_STRING);
+      expectOneOf(VALUE_STRING, VALUE_NUMBER_FLOAT, VALUE_NUMBER_INT);
```

Comment:
I couldn't enable lenient mode using gson builder. Is there an internal setting somewhere ?

The original problem is that [JsonElement TypeAdapter](https://github.com/google/gson/blob/master/gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java#L704) does the following optimization: 

```java
      case NUMBER:
        String number = in.nextString();
        return new JsonPrimitive(new LazilyParsedNumber(number));
```